### ISSUE TITLE
There is no reward for being caught by ghost in pacman_DDQN_agents.py

### ISSUE THREAD

**Issue by sjlee-me** [knowblesse/PacmanDQN#1]
A reward is calculated on pacman_DDQN_agents.py in branch feature/simpleDQN.
But it seems that it does not have a reward for case when an agent got caught by a ghost, while the original pacmanDQN code has a reward for the case.
Please check if you coded in the way you wanted.

This is a reward calculation in pacman_DDQN_agents.py, line 123.
```python
if (state.getFoodLevel() - self.lastState.getFoodLevel()) > 0 :
    # food eaten
    reward = 100
elif any(2 in _ for _ in state_array):
    # ghost in sight
    reward = -5
else:
    reward = -1
```

This is a reward calculation in pacmanDQN_Agents.py, line 144.
```python
# Process current experience reward
self.current_score = state.getScore()
reward = self.current_score - self.last_score
self.last_score = self.current_score

if reward > 20:
    self.last_reward = 50.    # Eat ghost   (Yum! Yum!)
elif reward > 0:
    self.last_reward = 10.    # Eat food    (Yum!)
elif reward < -10:
    self.last_reward = -500.  # Get eaten   (Ouch!) -500
    self.won = False
elif reward < 0:
    self.last_reward = -1.    # Punish time (Pff..)
```

**Comment by knowblesse** [comment]
@SlyJabiru You're right. The current _**pacman_DDQN_agents.py**_ code definitely lacks the punishment after being caught by the ghost. I'll add some lines in the **_PacmanDDQN.final()_** method to punish the agent. 
BTW, I did not encounter this error because I've trained the agent in a simple layout without a ghost. I guess other factors are preventing the learning.

**Comment by sjlee-me** [comment]
And I just found that **_PacmanDDQN.final()_** make in `final` function, sorry, but the punishment score looks mild. It should be bigger.

And all the reward calculation should be in `observation_step` function, to satisfy coherence with original _pacmanDQN_Agents.py_ and to make all reward calculations be done in one function.

However, it is not an urgent task to move all reward calculation logics into `observation_step` function, because our pacman agent fails even without a ghost. So you can move your logic later. 👍

**Comment by knowblesse** [comment]
Thanks for the quick feedback!👍👍

FYI, These are some of the options that I have tried (and failed)
- Extreme food/ghost reward (incl. starvation to the death)
- Larger punishment during movement
- Give a reward if the agent gets closer to the food. (proportional to the distance to the food)
- Register multiple episodes to the replay memory, if the agent eats food or gets the high score

For the `observation_step` function, I'll also change the code for coherence.
